File: hasad/mobile/lib/features/farmers/domain/compensation.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'compensation.freezed.dart';
part 'compensation.g.dart';

@freezed
class Compensation with _$Compensation {
  const factory Compensation({
    required String id,
    required String clientId,
    required String damageReportId,
    required double calculatedAmount,
    required double approvedAmount,
    required String status,
    required String remarks,
    required String rowVersion,
  }) = _Compensation;

  factory Compensation.fromJson(Map<String, dynamic> json) =>
      _$CompensationFromJson(json);
}

File: hasad/backend/Hasad.Domain/Entities/Compensation.cs
namespace Hasad.Domain.Entities;

public class Compensation
{
    public Guid Id { get; set; }
    public Guid ClientId { get; set; }

    public Guid DamageReportId { get; set; }
    public DamageReport? DamageReport { get; set; }

    public decimal CalculatedAmount { get; set; }
    public decimal ApprovedAmount { get; set; }

    public string Status { get; set; } = "Draft"; // Draft, Calculated, Approved, Paid

    public string Remarks { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public byte[] RowVersion { get; set; } = Array.Empty<byte>();
}
